relentless/optimize/method.py

```diff
@@ -372,15 +470,36 @@ def max_iter(self, value):
 
     @property
     def step_size(self):
-        r"""float: The step size hyperparameter (:math:`\alpha`)."""
+        r"""float: The step size hyperparameter (:math:`\alpha`). Must be positive."""
         return self._step_size
 
     @step_size.setter
     def step_size(self, value):
         if value <= 0:
-            raise ValueError('The specified step size must be positive.')
+            raise ValueError('The step size must be positive.')
         self._step_size = value
 
+    @property
+    def grad_scale(self):
+        r"""float or :class:`~relentless._collections.KeyedArray`: The gradient
+        scaling parameter or parameters (:math:`\mathbf{b}`) keyed on the design
+        variables. Must be positive."""
+        return self._grad_scale
+
+    @grad_scale.setter
+    def grad_scale(self, value):
```

Comment:
I think we want to allow `grad_scale` to be a regular-old dict. Imagine the user only wants to scale *some* but not all variables... this would cause an issue because the `KeyedArray` you generate here will not have the same keys as the design variables.

I think the solution I would use is to apply the scale parameters when you loop over the design variables. Try to get it from the dict if it exists (and `scale` is a dict), set it to 1.0 if `scale` is a dict but it's missing the key, otherwise use the scalar value of `scale`.